firefox-install-demo.toml

```diff
@@ -9,95 +9,131 @@ start_date = "2021-11-01"
+[metrics.install_volume_win8_1]
 data_source = "firefox_installs"
-select_expression = "IF(LOGICAL_OR(succeeded), 1, 0) * 100"
+select_expression = "IF(os = 'Win8.1' AND LOGICAL_OR(succeeded), 1, 0) * 100"
 type = "scalar"
 
-[metrics.install_volume_by_os_version.statistics]
+[metrics.install_volume_win8_1.statistics]
 sum = {}
+mean = {}
+
+[metrics.install_volume_win10]
+data_source = "firefox_installs"
+select_expression = "IF(os = 'Win10' AND LOGICAL_OR(succeeded), 1, 0) * 100"
+type = "scalar"
+
+[metrics.install_volume_win10.statistics]
+sum = {}
+mean = {}
 
 [alerts]
 
-[alerts.install_success_rate]
+[alerts.install_success_rate_win]
+type = "threshold"
+metrics = [
+    "install_volume_win7",
+    "install_volume_win8_1",
+    "install_volume_win10"
+]
+min = [90]
+
+[alerts.install_success_rate_win8]
 type = "threshold"
 metrics = [
-    "install_success_rate"
+    "install_volume_win8",
 ]
-# This one number is applied to each OS, unfortunately. Ideally
-# we would have per-branch configuration, since each Windows version
-# has its own fairly steady success rate.
 min = [86]
 
+
 [data_sources]
 
 [data_sources.firefox_installs]
 from_expression = """
     (
-        WITH success_rate_per_branch AS (
-            SELECT 
-                DATE(submission_timestamp) AS submission_date,
-                document_id,
-                build_channel,
-                succeeded,
-                CASE 
-                    WHEN os_version LIKE '6.1%' THEN 'Win7'
-                    WHEN os_version LIKE '6.2%' THEN 'Win8'
-                    WHEN os_version LIKE '6.3%' THEN 'Win8.1'
-                    WHEN os_version LIKE '10%' THEN 'Win10'
-                    ELSE "other"
-                END as branch,
-            FROM mozdata.firefox_installer.install
-            WHERE 
-                build_channel = "release"
-                AND installer_type = 'stub'
-        )
         SELECT 
-        submission_date,
-        document_id,
-        build_channel,
-        succeeded,
-        STRUCT (    -- this is the structure opmon expects
-            [
-            STRUCT (
-                "firefox-install-demo" AS key,   -- dummy experiment/rollout slug to make opmon happy
-                STRUCT(branch AS branch) AS value
-            )
-            ] AS experiments
-        ) AS environment
-        FROM success_rate_per_branch
-        WHERE branch != "other"
+            DATE(submission_timestamp) AS submission_date,
+            document_id,
+            build_channel,
+            succeeded,
+            CASE 
+                WHEN os_version LIKE '6.1%' THEN 'Win7'
+                WHEN os_version LIKE '6.2%' THEN 'Win8'
+                WHEN os_version LIKE '6.3%' THEN 'Win8.1'
+                WHEN os_version LIKE '10%' THEN 'Win10'
+                ELSE "other"
+            END as os,
+        FROM mozdata.firefox_installer.install
+        WHERE 
+            build_channel = "release"
+            AND installer_type = 'stub'
```

Comment:
The data source can also be simplified since we don't need to create "fake" branches

Reply:
This is an excellent improvement! It means we can copy existing STMO queries nearly verbatim.